pyproject.toml

```diff
@@ -23,27 +23,27 @@ documentation = "https://docs.flexcompute.com/projects/tidy3d/en/latest/"
 python = ">=3.9,<3.12"
 pyroots = ">=0.5.0"
 xarray = ">=0.16.2"
-importlib-metadata = "^6.0.0"
+importlib-metadata = ">=6.0.0"
 h5netcdf = "1.0.2"
 h5py = "^3.0.0"
 rich = "<12.6.0"
 numpy = "<2"
 matplotlib = "*"
-shapely = "^2.0"
+shapely = ">=2.0"
```

Comment:
What is the difference between "^" and ">="?

Reply:
It actually has different meanings according to how it is used: https://python-poetry.org/docs/dependency-specification/#caret-requirements 

It means the higher minor version for a given major version in this context >=2, <3. Whereas >= just means the highest version from this point >=2 including 3.

Reply:
Note that I changed it to match this: https://github.com/flexcompute/tidy3d/blob/v2.5.2/requirements/basic.txt#L10C1-L10C13

Reply:
yea probably we should keep the `^`, and even apply it to things  like `numpy`, `shapely`, `importlib-metadata`.

Maybe best for the 2.7 change (with the less restrictive versions) though?

Reply:
Yep that sounds good, will make another PR for 2.7 on this